enhancements/machine-config/user-data-secret-managed.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Machines user-data managed by the MCO
+authors:
+  - "@runcom"
+reviewers:
+  - "@yuqi-zhang"
+  - "@cgwalters"
+  - "@enxebre"
+  - "@JoelSpeed"
+approvers:
+  - "@yuqi-zhang"
+  - "@cgwalters"
+  - "@enxebre"
+  - "@JoelSpeed"
+creation-date: 2020-06-09
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Machines user-data managed by the MCO
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+
+## Summary
+
+Currently the installer creates [pointer Ignition configs][1] which end up as [user][2] [data][3] secrets [referenced by the installer-generated Machine(Sets)][4].  That makes things like the Ignition v2 (spec 3) transition tricky.
+The key point is that the Secret holding the initial stub ignition is completely unmanaged after it's created by the installer.
+In order to upgrade to the new Ignition spec, the MCO needs a way to update the Secret to the new Ignition spec.
+
+
+[1]: https://github.com/openshift/installer/blob/093ca65398fe567bdf63322894496cebbe3d2ade/pkg/asset/ignition/machine/node.go#L30-L36
+[2]: https://github.com/openshift/installer/blob/093ca65398fe567bdf63322894496cebbe3d2ade/pkg/asset/machines/master.go#L161-L170
+[3]: https://github.com/openshift/installer/blob/093ca65398fe567bdf63322894496cebbe3d2ade/pkg/asset/machines/worker.go#L197-L205
+[4]: https://github.com/openshift/installer/blame/093ca65398fe567bdf63322894496cebbe3d2ade/docs/user/aws/install_upi.md#L231-L232
+
+## Motivation
```

Comment:
So the overall motivation for this change is making ignition upgrades completely transparent for users which is not possible atm?

Reply:
yes, since the secret is unmanaged, the moment we upgrade MCO/installer/RHCOS to the new ignition, we'll still have this secret at the old version which works for "upgraded" Machines and new clusters, but breaks if you have Machines with v3 bootimages (custom Machinesets and Machine config pools). By managing the secret, the MCO can make sure it goes to v3 for new bootimages and remain at v2 (old version) for old machines